### PR TITLE
DCS-358 Raising app insights event when case is handed over.

### PR DESCRIPTION
**Pull request by andrewrlee**
This event will contain information about the source and target of the case.
This should allow us to easily see frequency at which certain prison/LDUs handover cases - see which prisons are using the service and which ones aren't.

Also adding this information to the audit trail in case we want to surface this through the
app at later date.

Whilst doing this I noticed that we weren't capturing any audit info for DMs - we're now doing that.

### CODE REVIEW

File: server/services/notifications/notificationService.js [hunk 1]
```diff
@@ -24,15 +25,27 @@ module.exports = function createNotificationService(
   }
 
   async function sendCaOrDm({ transition, bookingId, token, licence, prisoner, user }) {
-    const submissionTarget = await prisonerService.getOrganisationContactDetails(transition.receiver, bookingId, token)
+    const { submissionTarget, source, target } = await prisonerService.getDestinations(
```
Comment (bell-pa) [inline server/services/notifications/notificationService.js]: const destinations = ?

File: server/services/notifications/notificationService.js [hunk 2]
```diff
@@ -24,15 +25,27 @@ module.exports = function createNotificationService(
...
     if (transition.type === 'dmToCaReturn') {
       await licenceService.removeDecision(bookingId, licence)
     }
 
-    auditEvent(user.username, bookingId, transition.type, submissionTarget)
+    await eventPublisher.raiseCaseHandover({
+      username: user.username,
+      bookingId,
+      transitionType: transition.type,
+      submissionTarget,
```
Comment (bell-pa) [inline server/services/notifications/notificationService.js]: ...destinations ?

File: server/index.js [hunk 3]
```diff
@@ -75,22 +77,24 @@ const caAndDmNotificationSender = createCaAndDmNotificationSender(
   config
 )
 
+const eventPublisher = new EventPublisher(audit, appInsights)
```
Comment (bell-pa) [inline server/index.js]: Good idea.

Comment (andrewrlee) [reply server/index.js]: Yes, I could have done this - was just being a bit more specific. I'll have a look in a follow up

**Pull request merged**